src/datepicker/datepicker.js

```diff
@@ -411,16 +424,15 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
   };
 
   this._refreshView = function() {
-    var year = this.activeDate.getFullYear(),
-      month = this.activeDate.getMonth(),
-      firstDayOfMonth = new Date(this.activeDate);
+    var year = this.activeDate.getFullYear();
+    var month = this.activeDate.getMonth();
 
+    var firstDayOfMonth = new Date(this.activeDate);
     firstDayOfMonth.setFullYear(year, month, 1);
 
-    var difference = this.startingDay - firstDayOfMonth.getDay(),
-      numDisplayedFromPreviousMonth = difference > 0 ?
-        7 - difference : - difference,
```

Comment:
WTF was that???

Reply:
weird, it was declared then overwritten?

Reply:
Oh god! It was just a bad declaration